GH-65 Fix "Paramters" typo

stepio opened the pull request:
*Issue #65*

Fixed a typo in method's naming, also did a search and fixed the same in a couple of comments.
**Warning:** as existing API methods are renamed, it's a breaking change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

stepio commented:
Sorry, my bad, this almost duplicates #54, although I found and fixed one more occurrence in comments.

mukhtar commented:
Thanks for the fix. (de)serialization doesn't seem to be impacted by the typos, however, it would be preferable to keep the existing methods with the typos for backwards compatibility, mark them as deprecated, and add two new methods with the fixes.

stepio commented:
Thanks for feedback, you're right, it's safer to keep backward compatibility this way. Updated & force-pushed. Also slightly updated `.gitignore` file: added maven build folder and IntelliJ IDEA metadata files and folders.
